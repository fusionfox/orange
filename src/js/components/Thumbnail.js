import React from 'react'
import { Link } from 'react-router-dom'
import '../../css/components/Thumbnail.css';
import classNames from 'classnames';

const Thumbnail = (props) => {
  return (
    <li className={classNames(props.className, 'Thumbnail')}>
      <Link to={`/asset/${props.image.id}`}>
        <img className='Thumbnail__image' src={props.image.thumbnail} alt={props.image.title} />
      </Link>
    </li>
  )
}

export default Thumbnail
