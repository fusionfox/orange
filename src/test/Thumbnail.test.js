import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Thumbnail from '../js/components/Thumbnail'

test('Image Thumbnail rendered correctly', () => {
  const image = {
    id: 'PIA00231',
    title: 'Photo of the moon',
    description: 'This is a photo that we took of the moon',
    thumbnail: `https://images-assets.nasa.gov/image/PIA00231/PIA00231~thumb.jpg`,
    large: `https://images-assets.nasa.gov/image/PIA00231/PIA00231~large.jpg`
  }

  const component = renderer.create(
    <MemoryRouter>
      <Thumbnail image={image} />
    </MemoryRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
