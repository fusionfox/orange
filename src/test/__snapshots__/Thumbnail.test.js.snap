// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Thumbnail rendered correctly 1`] = `
<li
  className="Thumbnail"
>
  <a
    href="/asset/PIA00231"
    onClick={[Function]}
  >
    <img
      alt="Photo of the moon"
      className="Thumbnail__image"
      src="https://images-assets.nasa.gov/image/PIA00231/PIA00231~thumb.jpg"
    />
  </a>
</li>
`;
